main.go

```diff
@@ -2,152 +2,87 @@ package main
 
 import (
 	"fmt"
-	"strings"
+	"net/http"
+	"os"
+	"time"
 
 	"github.com/plaid/plaid-go/plaid"
 )
 
-// main contains example usage of all functions
-func main() {
-	// Create a Plaid client
-	client := plaid.NewClient("test_id", "test_secret", plaid.Tartan)
-
-	// GET /institutions
-	res, err := client.GetInstitutions(plaid.Tartan, nil, 50, 0)
-	if err != nil {
-		fmt.Println(err)
-	}
-	fmt.Println(res[0].Name, "has type:", res[0].Type)
-
-	// GET /institutions/5301a93ac140de84910000e0
-	inst, err := plaid.GetInstitution(plaid.Tartan, "5301a93ac140de84910000e0")
+func handleError(err error) {
 	if err != nil {
-		fmt.Println("Institution Error:", err)
+		panic(err)
 	}
-	fmt.Println(inst.Name, "has mfa:", strings.Join(inst.MFA, ", "))
-
-	// GET /categories
-	categories, err := plaid.GetCategories(plaid.Tartan)
-	if err != nil {
-		fmt.Println(err)
-	}
-	fmt.Println("First category:", categories[0])
-
-	// GET /categories/13001001
-	category, err := plaid.GetCategory(plaid.Tartan, "13001001")
-	if err != nil {
-		fmt.Println(err)
-	}
-	fmt.Println("category", category.ID, "is", strings.Join(category.Hierarchy, ", "))
-
-	// POST /auth
-	postRes, mfaRes, err :=
-		client.AuthAddUser("plaid_test", "plaid_good", "", "citi", nil)
-	if err != nil {
-		fmt.Println(err)
-	} else if mfaRes != nil {
-		switch mfaRes.Type {
-		case "device":
-			fmt.Println("--Device MFA--")
-			fmt.Println("Message:", mfaRes.Device.Message)
-		case "list":
-			fmt.Println("--List MFA--")
-			fmt.Println("Mask:", mfaRes.List[0].Mask, "\nType:", mfaRes.List[0].Type)
-		case "questions":
-			fmt.Println("--Questions MFA--")
-			fmt.Println("Question:", mfaRes.Questions[0].Question)
-		case "selections":
-			fmt.Println("--Selections MFA--")
-			fmt.Println("Question:", mfaRes.Selections[1].Question)
-			fmt.Println("Answers:", mfaRes.Selections[1].Answers)
-		}
-
-		postRes2, mfaRes2, err := client.AuthStepSendMethod(mfaRes.AccessToken, "type", "email")
-		if err != nil {
-			fmt.Println("Error submitting send_method", err)
-		}
-		fmt.Printf("%+v\n", mfaRes2)
-		fmt.Printf("%+v\n", postRes2)
-
-		postRes2, mfaRes2, err = client.AuthStep(mfaRes.AccessToken, "tomato")
-		if err != nil {
-			fmt.Println("Error submitting mfa", err)
-		}
-		fmt.Printf("%+v\n", mfaRes2)
-		fmt.Printf("%+v\n", postRes2)
-	} else {
-		fmt.Println(postRes.Accounts)
-		fmt.Println("Auth Get")
-		fmt.Println(client.AuthGet("test_citi"))
+}
 
-		fmt.Println("Auth DELETE")
-		fmt.Println(client.AuthDelete("test_citi"))
+func main() {
+	// Creates a new Plaid Client
+	clientOptions := plaid.ClientOptions{
+		os.Getenv("PLAID_CLIENT_ID"),
+		os.Getenv("PLAID_SECRET"),
+		os.Getenv("PLAID_PUBLIC_KEY"),
+		plaid.Sandbox,
+		&http.Client{},
+		plaid.LatestAPIVersion,
```

Comment:
Hmm what is the reason for allowing a version argument?

Reply:
⚡️